File: backend/hackathon/src/main/java/com/dev_trails/hackathon/Claim.java
package com.dev_trails.hackathon;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Column;
@Entity
public class Claim {
@Id
@GeneratedValue(strategy=GenerationType.IDENTITY)
public Long id;
@Column(unique=true)
public String claimNumber;
public String policyRef;
public Long riderId;
public String riderName;
public String product;
public String fraudRisk;
public String dateOfLoss;
public String status;
public String triggerType;
public Double amount;
public String zone;
public String approvedAt;
}

File: backend/models/Claim.js
const mongoose = require('mongoose');

const ClaimSchema = new mongoose.Schema({
  claim_number: { type: String, unique: true, sparse: true },
  policy_ref: { type: String },
  rider_id: { type: mongoose.Schema.Types.Mixed },
  rider_name: { type: String },
  product: { type: String },
  fraud_risk: { type: String },
  date_of_loss: { type: String },
  status: { type: String, default: 'PENDING' },
  trigger_type: { type: String },
  amount: { type: Number },
  zone: { type: String },
  approved_at: { type: String },
}, { timestamps: false });

ClaimSchema.set('toJSON', { virtuals: true });
ClaimSchema.virtual('id').get(function () { return this._id.toHexString(); });

module.exports = mongoose.model('Claim', ClaimSchema);
